kong/templates/nginx_kong.lua

```diff
@@ -45,8 +45,10 @@ lua_ssl_verify_depth ${{LUA_SSL_VERIFY_DEPTH}};
 > end
 
 # injected nginx_http_* directives
-> for k, v in pairs(nginx_http_directives)  do
-$(k) $(v);
+> for key, values in pairs(nginx_http_directives)  do
+>   for _, value in pairs(values) do
```

Comment:
Better use an array structure (of which we can control its order) and use a simpler loop in the template:
```
> for _, el in ipairs(nginx_http_directives) do
$(el.name) $(el.value);
> end
```

Reply:
To make sure I understand you correctly, you are asking to change the current layout from:
```lua
local nginx_http_directives = {
  lua_shared_dict = { "custom_cache1 1m", "custom_cache2 1m" }
}
```

to

```lua
local nginx_http_directives = { 
  {
    name = "lua_shared_dict",
    value = "custom_cache1 1m"
  }, 
  {
    name = "lua_shared_dict",
    value = "custom_cache2 1m"
  } 
}
```

Reply:
Yes, or any other layout you may think of that results in a simpler change in the template and allows ordering.

Reply:
IMHO the current implementation makes it easy to sort at two levels (first name, second value) easily but results in a slightly more complicated template.
The proposed approach makes reading all values of the same directive a little more complicated as well.

Willing to make the changes but would like to know why nested loops in the template are being considered complicated?

Reply:
> Willing to make the changes but would like to know why nested loops in the template are being considered complicated?

The more Lua mixed in the template, the more complicated to maintain for users? Can we implement the proposed solution? Thanks.

Reply:
The table could contain both an array and a hash part, I don't think the goals of simplifying the template, allowing the sort of directives, and the retrieval of values for recurrences of a directive get in the way of each other. Plus the latter can be done in the configuration table, and the former (array of directives) can be built especially for the purpose of compiling the nginx template.

Reply:
Yes, they are not going to get in each other's way. To keep things small, I'll update the PR to switch the data layout from hash to array of directives.

Reply:
@thibaultcha I've updated the code with the recent push.